source/kaleidic/mongo_standalone.d

```diff
@@ -887,8 +887,10 @@ struct SendBuffer {
 	private void size(size_t addition) {
 		if(buffer is null)
 			buffer = backing[];
-		if(position + addition > buffer.length)
-			buffer.length = buffer.length * 2;
+		if(position + addition > buffer.length) {
+                        buffer.length = buffer.length * 2;
+                        size(addition);
```

Comment:
if unlucky with huge data this recursion could lead to a stackoverflow (though very unlikely), this could be easily changed to a while loop without changes to make this not recursive. I think it would be better style because it would also make extending the size function with more code after the loop not risk being run multiple times if one misses this recursion.